### DISCUSSION TITLE
公共服务器套了Cloudflare后的注意事项

### DISCUSSION THREAD

**Discussion by xfgryujk** [xfgryujk/blivechat#180] [公告]
为了防御最近每天发生的DDoS攻击，我给chat.bilisc.com套了Cloudflare，同时会对请求频率做限制。计算请求频率时我的预期是一个房间最多开5个连接，因为如果超过了，B站自己也会断开连接。超过请求频率可能会有这样的表现：后台日志出现`ConnectionResetError: [WinError 64] 指定的网络名称不再可用`；或者在浏览器访问chat.bilisc.com时出现`Sorry, you have been blocked`

你可以采取以下措施降低请求频率：

* 在本地使用，并且开启`通过服务器转发消息`，这样每个房间只占用一条连接。注意每个房间页面都需要开启这个选项，否则没用。如果不是本地使用就没必要开这个选项了，否则会消耗我服务器的资源
* ~~如果直接用chat.bilisc.com，而不是本地使用，应该用身份码连接房间，而不要用房间ID。因为这样获取用户头像时会产生一个请求~~ v1.9.1已修改
* OBS浏览器源设置里勾选`当不可见时关闭源`，这样在后台的房间页面不会产生不必要的请求
* 升级到最新版，因为旧版连接房间失败时会每隔1秒无限重试。这是不合理的，会产生大量不必要的请求

另外，如果Cloudflare被墙或请求超时（出现`TimeoutError`），可以参考[临时解决方案](https://github.com/xfgryujk/blivechat/discussions/187)

**Comment by SSSSSSSan** [comment]
通过docker部署自己的弹幕姬服务器会收到这个改动的影响么，这样子使用弹幕姬又能给多少个用户提供服务捏？

**Reply by xfgryujk** [reply]
要不你改一个带cookie的版本

**Reply by SSSSSSSan** [reply]
嗯？我就是想问我用docker假设了自己的弹幕姬给别人用会收到这个每个房间5连接数的限制么，还是整个服务器会被限制为5个链接，或者不受任何限制？

**Reply by xfgryujk** [reply]
> 嗯？我就是想问我用docker假设了自己的弹幕姬给别人用会收到这个每个房间5连接数的限制么，还是整个服务器会被限制为5个链接，或者不受任何限制？

是按IP限制的，也就是整个服务器

**Reply by SSSSSSSan** [reply]
感谢解答

**Reply by SSSSSSSan** [reply]
如果配置了自己的开发者id和密钥是不是就不受公共服务器qos的限制了，因为我的服务器用的人还挺多的QAQ

**Reply by xfgryujk** [reply]
> 如果配置了自己的开发者id和密钥是不是就不受公共服务器qos的限制了，因为我的服务器用的人还挺多的QAQ

配置了就和我的服务器没关系了。但是你要上架幻星，否则只能在自己的直播间用

**Reply by YumeMichi** [reply]
我看了您配置文件里除了id和密钥还需要应用id，这个好像只对企业用户开放，个人用户申请不了

**Reply by xfgryujk** [reply]
> 我看了您配置文件里除了id和密钥还需要应用id，这个好像只对企业用户开放，个人用户申请不了

你看我像有企业的人吗

**Reply by SSSSSSSan** [reply]
审核好慢QwQ

**Comment by YumeMichi** [comment]
> > 我看了您配置文件里除了id和密钥还需要应用id，这个好像只对企业用户开放，个人用户申请不了
> 
> 你看我像有企业的人吗

sorry我进错【应用管理】了，我进成这个才需要企业用户（）
![](https://imgur.com/download/Rb90SmC/)
现在可以正常使用了，感谢
